File: Template/Vairuotojas.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Template
{
    public abstract class Vairuotojas
    {
        protected abstract void Vaziuoti(int km, double kuroDaugiklis);
        public abstract double LikutisProc();
        public abstract int LikutisKm();
        public void Vairuoti(int km)
        {
            Vaziuoti(km, 1);
        }
    }
}

File: Scala/src/Vairuotojas.scala
abstract class Vairuotojas {
  def Vaziuoti(km: Int, kuroDaugiklis : Double)
  def LikutisProc() : Double
  def LikutisKm() : Int
  def Vairuoti(km : Int)
  {
    Vaziuoti(km, 1)
    Console.println("Pravaziavo " + km + " km")
    Console.println("Liko " + LikutisProc() + "% bako")
    Console.println("Dar galima nuvaziuoti " + LikutisKm() + "km")
    Console.println("")
  }
}
